homeassistant/components/wittiot/__init__.py

```diff
@@ -0,0 +1,49 @@
+"""The WittIOT integration."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import CONF_IP, DOMAIN
+from .coordinator import WittiotDataUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+
+PLATFORMS = [Platform.SENSOR]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up WittIOT from a config entry."""
+
+    hass.config_entries.async_update_entry(entry, unique_id=entry.data[CONF_IP])
+    coordinator = WittiotDataUpdateCoordinator(
+        hass,
+        entry,
```

Comment:
The coordinator will have a `ConfigEntry` in `self.config_entry` as soon as you call `super.init()` so no need to pass it